### PR TITLE
Add optional CPU limit to OCI runtime.

### PR DESCRIPTION
**Pull request by vadimberezniker**
We can use this to set an upper bound on CPUs so that the CPU count looks uniform regardless of the underlying hardware.

https://github.com/buildbuddy-io/buildbuddy-internal/issues/3853

**Related issues**: N/A

### CODE REVIEW

File: enterprise/server/remote_execution/containers/ociruntime/ociruntime.go [hunk 1]
```diff
@@ -639,6 +640,15 @@ func (c *ociContainer) createSpec(ctx context.Context, cmd *repb.Command) (*spec
 	if err != nil {
 		return nil, fmt.Errorf("get container user: %w", err)
 	}
+
+	cpuSpecs := &specs.LinuxCPU{}
+	if *cpuLimit != 0 {
+		cpuSpecs = &specs.LinuxCPU{
+			Quota:  pointer(int64(*cpuLimit * 100_000)),
+			Period: pointer(uint64(100_000)),
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/containers/ociruntime/ociruntime.go:648]: a little comment about what units we're using here would be helpful for the reader

File: enterprise/server/remote_execution/containers/ociruntime/ociruntime.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ var (
 	runtimeRoot    = flag.String("executor.oci.runtime_root", "", "Root directory for storage of container state (see <runtime> --help for default)")
 	imageCacheRoot = flag.String("executor.oci.image_cache_root", "", "Root directory for cached OCI images. Defaults to './executor/oci/images' relative to the configured executor.root_directory")
 	pidsLimit      = flag.Int64("executor.oci.pids_limit", 2048, "PID limit for OCI runtime. Set to -1 for unlimited PIDs.")
+	cpuLimit       = flag.Int("executor.oci.cpu_limit", 0, "Hard limit for CPU resources, expressed as CPU count. Default is no limit.")
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/containers/ociruntime/ociruntime.go:51]: nit: Default (0) means no limit

**Review by tylerwilliams — APPROVED** [review]

File: enterprise/server/remote_execution/containers/ociruntime/ociruntime.go (hunk 1 above)
Comment (bduffany) [inline enterprise/server/remote_execution/containers/ociruntime/ociruntime.go:648]: nit: might be a little clearer as

```suggestion
	if *cpuLimit != 0 {
		period := 100*time.Millisecond	
		cpuSpecs = &specs.LinuxCPU{
			Quota:  pointer(int64(*cpuLimit) * period.Microseconds())),
			Period: pointer(uint64(period.Microseconds())),
```

**Review by bduffany — APPROVED** [review]

File: enterprise/server/remote_execution/containers/ociruntime/ociruntime.go (hunk 1 above)
Comment (vadimberezniker) [reply enterprise/server/remote_execution/containers/ociruntime/ociruntime.go]: fixed

Comment (vadimberezniker) [reply enterprise/server/remote_execution/containers/ociruntime/ociruntime.go]: done, thanks

File: enterprise/server/remote_execution/containers/ociruntime/ociruntime.go (hunk 2 above)
Comment (vadimberezniker) [reply enterprise/server/remote_execution/containers/ociruntime/ociruntime.go]: done

**Pull request merged**